wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift

```diff
@@ -123,7 +123,8 @@ public final class ZMUserSession: NSObject {
 
     let earService: EARServiceInterface
 
-    public var appLockController: AppLockType
+    public internal(set) var appLockController: AppLockType
+    private let contextStorage = LAContextStorage()
```

Comment:
Can I assume that the storage is correctly deallocated when the user logsout?
Otherwise I am not sure if `func tearDown()` is the right choice to release the store, but then the contextStorage needs to be optional.

Reply:
Yes, we can assume  that ZMUserSession gets deallocated along with all its objects. I'm not a fan of the tearDown pattern and I hope we can get rid of it.